packages/docusaurus-plugin-vercel-analytics/src/analytics.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import {inject} from '@vercel/analytics';
+
```

Comment:
We don't have a good way to forward plugin options to the client module here

The docs mention 3 options:
- mode
- debug
- beforeSend

https://vercel.com/docs/analytics/package

It will be complicated to support the unserializable beforeSend method.

However mode/debug are serializable so should be easier to support even though the solutions are a bit awkward.


### Use global data

```ts
    contentLoaded({actions}) {
      actions.setGlobalData(options);
    },
```

```ts
import globalData from '@generated/globalData';

const {debug, mode} = globalData["docusaurus-plugin-vercel-analytics"]["default"]
```

This assumes that the plugin uses the default id, in which case we probably want to forbid the usage of another id by throwing an error (it probably doesn't make sense to use multiple instances of this plugin anyway)

### Use client-side env variables

The PWA plugin does something a bit similar to pass options to `registerSw.ts`

<img width="534" alt="CleanShot 2024-01-04 at 16 43 11@2x" src="https://github.com/facebook/docusaurus/assets/749374/7e675eeb-3b1b-4499-86c4-3d9608df1fb8">